### PR TITLE
ci: use orka

### PR DESCRIPTION
**Pull request by v1v**
## Motivation/summary

Use Orka VMs and test how it goes in the CI 

## Issue

Test failure reported in https://github.com/elastic/apm-server/issues/8178

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 @Library('apm@current') _
 
 pipeline {
-  agent { label 'linux && immutable' }
+  agent { label 'orka && darwin' }
```
Comment (v1v) [inline Jenkinsfile:5]:
```suggestion
  agent { label 'orka && darwin && x86_64' }
```

File: Jenkinsfile [hunk 2]
```diff
@@ -228,7 +229,7 @@ pipeline {
         Build on a mac environment.
         */
         stage('OSX build-test') {
-          agent { label 'macosx && x86_64' }
+          agent { label 'orka && darwin' }
```
Comment (v1v) [inline Jenkinsfile:232]:
```suggestion
          agent { label 'orka && darwin && x86_64' }
```

**Comment by v1v** [comment]
### Test failures

image

https://apm-ci.elastic.co/job/apm-server/job/apm-server-mbp/job/PR-8142/122/testReport/junit/github/com_elastic_apm-server_x-pack_apm-server_sampling/TestStorageGC/

```
=== RUN   TestStorageGC
    processor_test.go:621: timed out waiting for value log garbage collection
--- FAIL: TestStorageGC (13.08s)
```

@elastic/apm-server , I need your help here, I've just seen the above test failure while running the build/test on MacOS, those MacOS workers are ephemeral rather than static workers, as we are in the transition to use a different provider. Would you mind if I ask you to look at the above test failure? 🙇

File: Jenkinsfile [hunk 3]
```diff
@@ -53,7 +54,7 @@ pipeline {
       }
       options { skipDefaultCheckout() }
       steps {
-        pipelineManager([ cancelPreviousRunningBuilds: [ when: 'PR' ] ])
+        //pipelineManager([ cancelPreviousRunningBuilds: [ when: 'PR' ] ])
```
Comment (v1v) [inline Jenkinsfile:57]:
```suggestion
        pipelineManager([ cancelPreviousRunningBuilds: [ when: 'PR' ] ])
```

File: Jenkinsfile [hunk 4]
```diff
@@ -28,18 +28,19 @@ pipeline {
...
   triggers {
     issueCommentTrigger("(${obltGitHubComments()}|^run\\W+(?:the\\W+)?(hey-apm|package|arm|windows)\\W+tests|^/test|^/hey-apm|^/package|^/test windows)")
+    cron 'H/15 * * * *'
   }
   parameters {
     booleanParam(name: 'Run_As_Main_Branch', defaultValue: false, description: 'Allow to run any steps on a PR, some steps normally only run on main branch.')
-    booleanParam(name: 'arm_ci', defaultValue: true, description: 'Enable ARM build')
-    booleanParam(name: 'linux_ci', defaultValue: true, description: 'Enable Linux build')
+    booleanParam(name: 'arm_ci', defaultValue: false, description: 'Enable ARM build')
+    booleanParam(name: 'linux_ci', defaultValue: false, description: 'Enable Linux build')
```
Comment (v1v) [inline Jenkinsfile:36]:
```suggestion
    booleanParam(name: 'arm_ci', defaultValue: true, description: 'Enable ARM build')
    booleanParam(name: 'linux_ci', defaultValue: true, description: 'Enable Linux build')
```

File: Jenkinsfile [hunk 5]
```diff
@@ -28,18 +28,19 @@ pipeline {
...
-    booleanParam(name: 'intake_ci', defaultValue: true, description: 'Enable test')
-    booleanParam(name: 'test_ci', defaultValue: true, description: 'Enable test')
-    booleanParam(name: 'test_sys_env_ci', defaultValue: true, description: 'Enable system and environment test')
-    booleanParam(name: 'bench_ci', defaultValue: true, description: 'Enable benchmarks')
-    booleanParam(name: 'release_ci', defaultValue: true, description: 'Enable build the release packages')
+    booleanParam(name: 'intake_ci', defaultValue: false, description: 'Enable test')
+    booleanParam(name: 'test_ci', defaultValue: false, description: 'Enable test')
+    booleanParam(name: 'test_sys_env_ci', defaultValue: false, description: 'Enable system and environment test')
+    booleanParam(name: 'bench_ci', defaultValue: false, description: 'Enable benchmarks')
+    booleanParam(name: 'release_ci', defaultValue: false, description: 'Enable build the release packages')
```
Comment (v1v) [inline Jenkinsfile:43]:
```suggestion
    booleanParam(name: 'intake_ci', defaultValue: true, description: 'Enable test')
    booleanParam(name: 'test_ci', defaultValue: true, description: 'Enable test')
    booleanParam(name: 'test_sys_env_ci', defaultValue: true, description: 'Enable system and environment test')
    booleanParam(name: 'bench_ci', defaultValue: true, description: 'Enable benchmarks')
    booleanParam(name: 'release_ci', defaultValue: true, description: 'Enable build the release packages')
```

File: Jenkinsfile [hunk 6]
```diff
@@ -28,18 +28,19 @@ pipeline {
   }
   triggers {
     issueCommentTrigger("(${obltGitHubComments()}|^run\\W+(?:the\\W+)?(hey-apm|package|arm|windows)\\W+tests|^/test|^/hey-apm|^/package|^/test windows)")
+    cron '@hourly'
```
Comment (v1v) [inline Jenkinsfile:31]:
```suggestion
```

**Comment by v1v** [comment]
@marclop regarding your question https://github.com/elastic/apm-server/pull/8032#issuecomment-1132421919

in this case, the `build`/`test` takes in total 6 minutes approx:

image

**Comment by marclop** [comment]
@v1v so using okra is much faster than our current approach for MacOSX?

**Comment by v1v** [comment]
> @v1v so using okra is much faster than our current approach for MacOSX?

This [discover url](https://ci-stats.elastic.co/goto/cdc94260-d813-11ec-b3b3-7dc6fc35bc94) filters by OSX stage in the last 20 days that didn't fail:

image

It's a bit faster. Though, I didn't search for failed stages, that maybe could be the ones with issues

Unfortunately the static workers can easily get corrupted and then things start going slow, when it happens, then it's required to contact the CI System team since they own those machines. The good thing, static workers is now the past, and ephemeral workers with Orka is the present, hence the reason for this PR

**Review by kuisathaverat — APPROVED** [review]

**Review by axw — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

**Comment by axw** [comment]
@v1v TestStorageGC is now passing on orka.

**Pull request merged**